x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/AutodetectMemoryLimitIT.java

```diff
@@ -225,12 +229,12 @@ public void testManyDistinctOverFields() throws Exception {
         // Assert we haven't violated the limit too much
         GetJobsStatsAction.Response.JobStats jobStats = getJobStats(job.getId()).get(0);
         ModelSizeStats modelSizeStats = jobStats.getModelSizeStats();
-        assertThat(modelSizeStats.getModelBytes(), lessThan(120500000L));
-        assertThat(modelSizeStats.getModelBytes(), greaterThan(70000000L));
+        assertThat(getEffectiveModelSize(modelSizeStats.getModelBytes()), lessThan(ByteSizeValue.ofMb(memoryLimitMb).getBytes() * 1.05));
         assertThat(modelSizeStats.getMemoryStatus(), equalTo(ModelSizeStats.MemoryStatus.HARD_LIMIT));
```

Comment:
The reporting of memory usage is different on different plattforms: on Linux and Windows we report actual process memory usage, while on MacOS this information is not available and hence we report estimated memory usage. This makes the test for a specific memory limit brittle. To improve robustness, I only check that the job is in the `hard_limit` state.